### DISCUSSION TITLE
Operation failed due to object-level permissions violation when bulk editing interfaces MACs

### DISCUSSION THREAD

**Discussion by gabrielmarmen** [netbox-community/netbox#16222] [Help Wanted!]
I am currently in the process of importing some devices and I have an issue when I try to bulk edit interfaces MAC addresses using a csv. I get the following error :  **Operation failed due to object-level permissions violation**

To proceed with the import I did the following : 
- Create a device type that corresponds to the devices I'm trying to import
- Create a csv with the devices 
- Create a csv with the ips (Each ip matching with the devices name and interface)
- Create a csv with the interfaces ids and the corresponding MAC address (For updating the existing interfaces

I tried the following:
- Edit only one item using the csv : Not working
- Editing the item manually using the GUI : Working
- Creating permissions that have absolutely every rights : Not Working

Any ideas as to why I am getting this error ? Is there anywhere I could get more details on this error ?

Thanks !

**Comment by candlerb** [comment]
What Netbox version are you running?

Can you show an example of the CSVs you're uploading? (Header row + at least one data row)

Have you tried logging in as an "admin" user?

Aside: the error you show is `PermissionsViolation` and it's raised in various places in `netbox/netbox/views/generic/bulk_views.py` if you want to insert some debugging statements.

**Reply by gabrielmarmen** [reply]
I am currently logged in as a superuser. The main admin account.

I'm running v3.7.3

Here is an example of the csv:
[interfaces_update - Copy.csv](https://github.com/netbox-community/netbox/files/15394266/interfaces_update.-.Copy.csv)

I don't know enough to debug to be honest. Do you have any ressources I could use to get started with debugging?

**Reply by candlerb** [reply]
That CSV upload works for me, with Netbox v3.7.8, after changing the ID to the ID of an interface that I have.

```
id,mac_address
24,7c-1e-b3-00-da-8e
```

![image](https://github.com/netbox-community/netbox/assets/44789/8ed9116f-9fe3-4e68-a0f1-4dfcddd0c6df)

EDIT: I tried both pasting in the data to the box, and uploading a file (I note it has Windows 0D 0A line endings, but that didn't cause a problem for me)

I suggest you upgrade to 3.7.8 first - I see some permission-related things in the [release notes](https://docs.netbox.dev/en/stable/release-notes/version-3.7/), although nothing that seems to relate directly to this particular issue.

**Reply by gabrielmarmen** [reply]
All right, thanks for the follow up. I will test that !
